graph/tags/tags.go

```diff
@@ -2,22 +2,20 @@ package tags
 
 import (
 	"fmt"
-	"regexp"
+
+	"github.com/docker/distribution/digest"
+	"github.com/docker/distribution/registry/api/v2"
 )
 
 const DEFAULTTAG = "latest"
 
-var (
-	//FIXME this regex also exists in registry/v2/regexp.go
-	validTagName = regexp.MustCompile(`^[\w][\w.-]{0,127}$`)
-)
-
 // ValidateTagName validates the name of a tag
 func ValidateTagName(name string) error {
 	if name == "" {
 		return fmt.Errorf("tag name can't be empty")
 	}
-	if !validTagName.MatchString(name) {
+
+	if !v2.TagNameRegexp.MatchString(name) && !digest.DigestRegexp.MatchString(name) {
```

Comment:
@calavera It is probably not. We are working on a typesafe package to handle this better. Let's focus on correctness, for nwo.